components/RegisterISCN.vue

```diff
@@ -0,0 +1,523 @@
+  author: iscnData.value.author.name,
+  authorDescription: iscnData.value.author.description,
+  license: iscnData.value.license,
+  contentFingerprints: iscnData.value.contentFingerprints.map(f => f.url),
+  inLanguage: iscnData.value.language,
+  publisher: iscnData.value.publisher,
+  isbn: iscnData.value.isbn,
+  datePublished: iscnData.value.publicationDate
+    ? new Date(iscnData.value.publicationDate).toISOString()
+    : undefined,
+  url: iscnData.value.contentFingerprints[0]?.url || '',
+  tagsString: iscnData.value.tags?.join(', ') || '',
+  ipfsHash: iscnData.value.ipfsHash,
+  arweaveId: iscnData.value.arweaveId,
+  sameAs: formattedSameAsList.value,
+  thumbnailUrl: iscnData.value.coverUrl
+}))
+
+const isFormValid = computed(() => {
+  const requiredFields = {
+    title: !!iscnData.value.title,
+    description: !!iscnData.value.description,
+    authorName: !!iscnData.value.author.name,
+    contentUrl: !!iscnData.value.contentFingerprints.some(f => !!f.url)
+  }
+
+  error.value = Object.entries(requiredFields)
+    .find(([_, isValid]) => !isValid)?.[0]
+    ?.toUpperCase() || ''
+
+  return Object.values(requiredFields).every(Boolean)
+})
+
+onMounted(() => {
+  const initialData = initializeFromSessionStorage()
+  if (initialData) {
+    iscnData.value = initialData
+  }
+  calculateISCNFee()
+})
+
+const initializeFromSessionStorage = () => {
+  if (process.server) {
+    return null
+  }
+
+  try {
+    const storedData = window.sessionStorage.getItem('uploadFileData')
+    if (!storedData) {
+      return null
+    }
+
+    const data: UploadFileData = JSON.parse(storedData)
+
+    const baseData = {
+      type: 'book',
+      title: data.epubMetadata?.title || '',
+      description: stripHtmlTags(data.epubMetadata?.description || ''),
+      isbn: '',
+      publisher: '',
+      publicationDate: '',
+      author: {
+        name: data.epubMetadata?.author || '',
+        description: '',
+        url: ''
+      },
+      license: 'All rights reserved',
+      contentFingerprints: [] as Array<{ url: string }>,
+      downloadableUrls: [] as Array<{
+        url: string;
+        type: string;
+        fileName: string;
+      }>,
+      coverUrl: `ar://${data.epubMetadata?.thumbnailArweaveId}`,
+      language: formatLanguage(data.epubMetadata?.language || ''),
+      tags: data.epubMetadata?.tags || [],
+      ipfsHash: [
+        data.thumbnailIpfsHash,
+        ...data.fileRecords.map(record => record.ipfsHash)
+      ]
+        .filter(Boolean),
+      arweaveId: data.fileRecords
+        .map(record => record.arweaveLink || record.arweaveId)
+        .filter(Boolean)
+    }
+
+    const downloadableFiles = data.fileRecords.filter(
+      record => record.fileType === 'epub' || record.fileType === 'pdf'
+    )
+
+    if (downloadableFiles.length) {
+      baseData.downloadableUrls = downloadableFiles.map(file => ({
+        url: file.arweaveLink || `ar://${file.arweaveId}`,
```

Comment:
`arweaveLink` should be use when user choose to opt in DRM (and preferably with encryption on)

Reply:
[🎨 Display Arweave link for encrypted content](https://github.com/likecoin/nft-book-press/pull/292/commits/2cdcdd0ad359177f0c5d927fd446d4b2b30335d1)